src/default_parameters.jl

```diff
@@ -113,8 +115,12 @@ The default values of the keywords define the default model setup.
     restart_id::Integer=1           # run_id of restart file in run????/restart.jld2
 end
 
-function nlev_default(model::Symbol)
-    model == :barotropic && return 1
-    model == :shallowwater && return 1
-    model == :primitive && return 8
+function nlev_default(model::Symbol, initialize_σ_manually::Bool, σ_half::Vector{T}) where T<:Real
+    if !(initialize_σ_manually)
+        model == :barotropic && return 1
+        model == :shallowwater && return 1
+        model == :primitive && return 8
+    else 
+        return length(σ_half) - 1
+    end
```

Comment:
Suggestion would be
```julia
function nlev_default(model::Symbol, σ_levels_half::AbstractVector)
    if length(σ_half)  == 0    # choose nlev automatically
        model == :barotropic && return 1
        model == :shallowwater && return 1
        model == :primitive && return 8
    else     # use manually initialized σ_levels_half instead
        @assert σ_levels_half[1] == 0 "First manually specified σ_levels_half has to be zero 0"
        @assert σ_levels_half[end] == 1 "Last manually specified σ_levels_half has to be 1."
        @assert isincreasing(σ_levels_half) "Manually specified σ_levels_half have to be increasing"
        return length(σ_levels_half) - 1
    end
end
```
That way
- `run_speedy(nlev=4)` would use `nlev=4` and ignore `model` and potentially also `σ_levels_half` if that's specified too.
- `run_speedy(σ_levels_half=[0,0.2,0.5,0.8,1])` would set `nlev=4` through the function above.

Reply:
if both are specified, it doesn't work anymore, I add an `@assert` for that case in the `vertical coordinates`